### PR TITLE
affy/justrma add stub, nf-test and snapshot

### PR DESCRIPTION
**Pull request by sateeshperi**

### CODE REVIEW

File: modules/nf-core/affy/justrma/environment.yml [hunk 1]
```diff
@@ -5,3 +5,4 @@ channels:
   - defaults
 dependencies:
   - bioconda::bioconductor-affy=1.78.0
+  - conda-forge::r-base=4.3.1
```
Comment (jfy133) [inline modules/nf-core/affy/justrma/environment.yml:8]: Doesn't that come with a bioconductor package!? Or is it unpinned in conda?

File: modules/nf-core/affy/justrma/main.nf [hunk 2]
```diff
@@ -21,5 +21,14 @@ process AFFY_JUSTRMA {
     task.ext.when == null || task.ext.when
 
     script:
+    prefix = task.ext.prefix ?: "${meta.id}"
     template 'affy_justrma.R'
+
+    stub:
+    """
+    touch test_eset.rds
+    touch test_matrix.tsv
```
Comment (jfy133) [inline modules/nf-core/affy/justrma/main.nf:30]:
```suggestion
    touch ${prefix}_eset.rds
    touch ${prefix}_matrix.tsv
```

File: modules/nf-core/affy/justrma/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            }
+        }
+    }
+
+    test("homo_sapiens genome affy_array [celfiles]") {
+
+        when {
+            process {
+                """
+                ch_samplesheet = Channel.of([ [ id:'test' ], file(params.test_data['homo_sapiens']['genome']['affy_array_samplesheet'], checkIfExists: true) ])
```
Comment (jfy133) [inline modules/nf-core/affy/justrma/tests/main.nf.test:32]: Please use new `test_data_path` notation thingy

Comment (jfy133) [reply modules/nf-core/affy/justrma/tests/main.nf.test:32]: (as in the rest)

File: modules/nf-core/affy/justrma/tests/nextflow.config [hunk 4]
```diff
@@ -0,0 +1,6 @@
+process {
+    withName: 'test_affy_justrma:AFFY_JUSTRMA' {
+        ext.args = '--sample_name_col name --build_annotation TRUE'
```
Comment (jfy133) [inline modules/nf-core/affy/justrma/tests/nextflow.config:3]: Where does this get parsed? I don't see any further reference to it

**Review by jfy133 — APPROVED** [review]
Generally OK, a few thigns I don't fuilly understand

File: modules/nf-core/affy/justrma/environment.yml (hunk 1 above)
Comment (SPPearce) [reply modules/nf-core/affy/justrma/environment.yml:8]: Correct, it is unpinned in conda, was using 4.3.3 in the conda recipe.

File: modules/nf-core/affy/justrma/tests/nextflow.config (hunk 4 above)
Comment (SPPearce) [reply modules/nf-core/affy/justrma/tests/nextflow.config:3]: It was being read for the second test, but then not actually used as the name of the process had changed from pytest. Actually using it now.

**Pull request merged**